internal/allocator/allocator.go

```diff
@@ -358,14 +356,16 @@ func poolCount(p *config.Pool) int64 {
 }
 
 // poolFor returns the pool that owns the requested IP, or "" if none.
-func poolFor(pools map[string]*config.Pool, ip net.IP) string {
+func poolFor(pools map[string]*config.Pool, ips []net.IP) string {
 	for pname, p := range pools {
-		if p.AvoidBuggyIPs && ipConfusesBuggyFirmwares(ip) {
-			continue
-		}
-		for _, cidr := range p.CIDR {
-			if cidr.Contains(ip) {
-				return pname
+		for _, ip := range ips {
+			if p.AvoidBuggyIPs && ipConfusesBuggyFirmwares(ip) {
+				continue
+			}
+			for _, cidr := range p.CIDR {
+				if cidr.Contains(ip) {
+					return pname
```

Comment:
Sorry I wasn't clear in my previous comment. Did not mean to remove the check on the number of ips which made a lot of sense since we are looking for a pool that have both ips.
What I meant was to exit from the inner loop (the `p.CIDR` one) if we had one match, but continue on the outer (the `ips` one) so we avoid the case of different CIDRs matching the same ip.
What could happen in that case is that we match twice a v4 address, we return (because found==len(ips) but the v6 address is not in the pool.